notebook/notebookapp.py

```diff
@@ -674,11 +676,16 @@ def _default_cookie_secret_file(self):
     def _default_cookie_secret(self):
         if os.path.exists(self.cookie_secret_file):
             with io.open(self.cookie_secret_file, 'rb') as f:
-                return f.read()
+                key =  f.read()
         else:
-            secret = encodebytes(os.urandom(1024))
-            self._write_cookie_secret_file(secret)
-            return secret
+            key = encodebytes(os.urandom(1024))
+            self._write_cookie_secret_file(key)
+        h = sha256(digest_size=len(key))
```

Comment:
I was going to make a comment here about what happens if the cookie file happened to be empty (leading to a 0 length), then I noticed that at least on my systems I can't set this keyword argument:

```ipython
In [1]: from hashlib import sha256

In [2]: sha256(digest_size=0)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-94e5dda9a83d> in <module>()
----> 1 sha256(digest_size=0)

TypeError: openssl_sha256() takes no keyword arguments
```

Reply:
hum, weird.